src/content_main.js

```diff
@@ -186,78 +222,57 @@ function getPaymentPointerFromPage() {
-			(paymentPointerOnPage !== cachedPaymentPointer);
-
-		if (isNewPaymentPointer) {
-			// Note that paymentPointerOnPage may be newly added to the current page and tab,
-			// but it is possible that this payment pointer was recently validated in another tab.
-			// Therefore it is necessary to use handleValidationAndSetExtensionIcon which checks the
-			// browser storage to see if the payment pointer has been recently validated.
-			handleValidationAndSetExtensionIcon(paymentPointerOnPage);
-			cachedPaymentPointer = paymentPointerOnPage;
-		}
-	}, NEW_PAYMENT_POINTER_CHECK_RATE_MS);
-}
-
 /**
  * Ensures that the extension icon is updated to the correct state when switching between tabs.
  */
-function handleUpdatingExtensionIconOnTabVisibilityChange() {
-	document.addEventListener('visibilitychange', (event) => {
+async function handleUpdatingExtensionIconOnTabVisibilityChange() {
+	document.addEventListener('visibilitychange', async (event) => {
 		const tabIsVisible = !document.hidden;
 
 		if (tabIsVisible) {
 			const paymentPointerOnPage = getPaymentPointerFromPage();
-			if (paymentPointerOnPage === null) {
-				setExtensionIconMonetizationState(false);
-			} else {
-				handleValidationAndSetExtensionIcon(paymentPointerOnPage);
-			}
+			const isValidPaymentPointer = await validatePaymentPointer(paymentPointerOnPage);
+
+			// Set the extension icon to monetized if the payment pointer is valid
+			setExtensionIconMonetizationState(isValidPaymentPointer);
 		}
 	});
 }
 
 /**
- * Given a payment pointer, if it is valid, this function sets the extension icon to its monetized
- * state and stores the payment pointer in browser storage, otherwise the extension icon is set to
- * its unmonetized state.
+ * Validate the payment pointer or check the recently validated result
+ * and return whether the payment pointer is valid or not. Stores the payment
+ * pointer if it does not already exist.
  *
- * @param {string} paymentPointer The payment pointer to validate.
- * @returns {Promise<>} A promise that resolves when validation is finished.
+ * @param {String} paymentPointer The payment pointer to validate.
+ * @returns {Promise<boolean>} The validated payment pointer or null if invalid.
  */
-async function handleValidationAndSetExtensionIcon(paymentPointer) {
-	if (await isPaymentPointerRecentlyValidated(paymentPointer)) {
-		setExtensionIconMonetizationState(true);
-	} else {
-		// Validate if not validated recently
-		if (await isPaymentPointerValid(paymentPointer)) {
-			// Store the payment pointer and the UTC timestamp for the time it was validated (now)
-			await storeDataIntoAkitaFormat({
-				paymentPointer: paymentPointer,
-				validationTimestamp: Date.now()
-			}, AKITA_DATA_TYPE.PAYMENT);
-			setExtensionIconMonetizationState(true);
+async function validatePaymentPointer(paymentPointer) {
+	let isValid = false;
+
+	if (paymentPointer) {
+		if (await isPaymentPointerRecentlyValidated(paymentPointer)) {
+			isValid = true;
```

Comment:
nit: I prefer early return i.e. `return true;`

Reply:
We should document this practice in #85